cfme/services/catalogs/catalog.py

```diff
@@ -126,56 +115,38 @@ def exists(self):
 class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='All Catalogs')
+    VIEW = CatalogsView
 
     def step(self):
         self.prerequisite_view.navigation.select('Services', 'Catalogs')
-        tree = accordion.tree('Catalogs')
-        tree.click_path('All Catalogs')
-
-    def resetter(self):
-        # Default list view
-        tb.select('List View')
-        if paginator.page_controls_exist():
-            # Make sure nothing is selected
-            sel.check(paginator.check_all())
-            sel.uncheck(paginator.check_all())
+        self.view.catalogs.tree.click_path("All Catalogs")
 
 
-@navigator.register(Catalog, 'Add')
-class Add(CFMENavigateStep):
+@navigator.register(Catalog, 'AddCatalog')
+class AddCatalog(CFMENavigateStep):
     prerequisite = NavigateToSibling('All')
 
-    def am_i_here(self):
-        return match_page(summary='Adding a new Catalog')
+    VIEW = AddCatalogView
 
     def step(self):
-        cfg_btn('Add a New Catalog')
+        self.view.configuration.item_select('Add a New Catalog')
 
 
 @navigator.register(Catalog, 'Details')
 class Details(CFMENavigateStep):
     prerequisite = NavigateToSibling('All')
 
-    def am_i_here(self):
-        # header text as separate var to reduce quote escaping
-        return match_page(summary='Catalog "{}"'.format(self.obj.name))
+    VIEW = DetailsCatalogView
 
     def step(self):
-        # Best effort to make sure the catalog is in the list
-        if paginator.page_controls_exist():
-            paginator.results_per_page(1000)
-        listview_table.click_row_by_cells(
-            {'Name': self.obj.name, 'Description': self.obj.description})
+        self.view.catalogs.tree.click_path("All Catalogs", self.obj.name)
```

Comment:
**REQUIRED**: Use `self.prerequisite_view` since to get to the location, you usually need the previous view and do something with it to get to the location this step represents. In this case it probably works because the views are inherited, but this leads to bad copypasting and it might also lead to pulled hair when something changes :)